R/LongitudinalQuantities.R

```diff
@@ -0,0 +1,214 @@
+
+#' @include DataJoint.R
+#' @include Quantities.R
+NULL
+
+
+#' `LongitudinalQuantities` Object & Constructor Function
+#'
+#' Constructor function to generate a `LongitudinalQuantities` object.
+#'
+#' @param object ([`JointModelSamples`]) \cr Samples as drawn from a Joint Model
+#'
+#' @param groups (`character`, `NULL`)\cr which patients to calculate the desired
+#' quantities for.
+#'
+#' @param time_grid (`numeric` or `NULL`)\cr a vector of time points to calculate the desired
+#' quantity at. If `NULL` will be set to `seq(0, max_longitudinal_time, length = 201)`
+#'
+#' @details
+#' Note that unlike [SurvivalQuantities], [LongitudinalQuantities] does not support
+#' group aggregation.
+#'
+#' @slot quantities (`Quantities`)\cr The sampled quantities. Should contain 1 element per
+#' element of `group`
+#' @slot groups (`list`)\cr See argument section for details
+#' @slot time_grid (`numeric`)\cr See argument section for details
+#' @slot data ([`DataJoint`])\cr The data that the Joint Model was fitted to to produce
+#' the samples/quantities
+#'
+#'
+#' @family LongitudinalQuantities
+#' @name LongitudinalQuantities-class
+#' @export LongitudinalQuantities
+.LongitudinalQuantities <- setClass(
+    "LongitudinalQuantities",
+    slots = c(
+        "quantities" = "Quantities",
+        "data" = "DataJoint",
+        "groups" = "list",
+        "time_grid" = "numeric"
+    )
+)
+
+#' @rdname LongitudinalQuantities-class
+LongitudinalQuantities <- function(
+    object,
+    groups = NULL,
+    time_grid = NULL
+) {
+    assert_that(inherits(object, "JointModelSamples"))
+    assert_that(is.null(groups) || is.character(groups))
+    data <- as.list(object@data)
+    patients <- decompose_patients(groups, names(data$pt_to_ind))
+    time_grid <- expand_time_grid(time_grid, max(data[["Tobs"]]))
+
+    gq <- generateQuantities(
+        object,
+        patients = patients$unique_values,
+        time_grid_lm = time_grid,
+        time_grid_sm = numeric(0)
+    )
+
+    quantities_raw <- extract_quantities(gq, type = "lm_identity")
+
+    quantities <- lapply(
+        patients$indexes,
+        average_samples_by_index,
+        time_index = seq_along(time_grid),
+        quantities = quantities_raw
+    )
+
+    .LongitudinalQuantities(
+        quantities = Quantities(quantities),
+        data = object@data,
+        groups = patients$groups,
+        time_grid = time_grid
+    )
+}
+
+
+
+
+#' `as.data.frame`
+#'
+#' @param x ([`LongitudinalQuantities`]) \cr Longitudinal Quantities
+#' @param ... Not used
+#' @family LongitudinalQuantities
+#' @export
+as.data.frame.LongitudinalQuantities <- function(x, ...) {
+    res <- as.data.frame(
+        x@quantities,
+        time_grid = x@time_grid,
+        groups = x@groups,
+        type = "na"
+    )
+    res["type"] <- NULL
+    return(res)
+}
+
+
+
+#' summary
+#'
+#'
+#' @description
+#' This method returns a `data.frame` of the longitudinal quantities
```

Comment:
```suggestion
#' This method returns a `data.frame` of the longitudinal quantities.
```
(I would use sentences in the descriptions, so ending with full stop)